octoprint_findmymrbeam/__init__.py

```diff
@@ -103,30 +107,52 @@ def _get_server_registry_data(self):
 					netifaces=self._get_netifaces()
 					)
 
-	@staticmethod
-	def _get_dns_domain_names():
+	def _get_dns_domain_names(self):
 		"""
 		Try and get the dns domain name for the device
+		Warning: This method will consume some time due to trying to verify dnsdomain names,
+		The implementation of this method is done so that the verification would only be done if it is needed
 
 		Returns:
-			list: Collected DNS domain names for the device, in case of error, return ["Exception ..."]
+			list: Collected & verified DNS domain names for the device, in case of error, return ["Exception ..."]
 
 		"""
-
 		try:
 			dns_domain_names_str = subprocess.check_output(["dnsdomainname", "-A"], stderr=subprocess.STDOUT)
 		except subprocess.CalledProcessError as e:
 			# The complete Error is copied to the return so that we can see it in findmrbeam registry
 			return ["Exception: {} ".format(e) + "##Code: {} ".format(e.returncode) + "##Output: {}".format(e.output)]
-		except (subprocess.SubprocessError, OSError) as e:
+		except (OSError, ValueError) as e:
 			# The complete Error is copied to the return so that we can see it in findmrbeam registry
 			return "Exception: {}".format(e)
 		else:
 			# Remove new line if any and Convert to a list of sub strings, each could be a dns domain name candidate
 			dns_domain_names = dns_domain_names_str.strip().split(" ")
 			# Remove duplicates
 			dns_domain_names = list(set(dns_domain_names))
-			return dns_domain_names
+
+		# If same list of names, return the last verified list
+		if dns_domain_names == self._dns_domain_names:
+			return self._verified_dns_domain_names
+
+		# Check if all retrieved domain names are reachable
+		self._dns_domain_names = dns_domain_names
+		verified_dns_domain_names = []
+		http_status_code = None
+		for d_name in dns_domain_names:
+			try:
+				http_status_code = urllib.urlopen("http://{}".format(d_name)).getcode()
```

Comment:
Could you add a timeout here? If the domain name work it should respond really fast as it is the same device. So a timeout will speed up the validation process

Reply:
This function doesn't have timeout for python2.7

Reply:
found another library liburl2 that supports timeout. did some testing and in my network it need min 0.8 sec to complete. not sure on other network setups how long it would take!

I will just add 3sec, but this will be a very wild guess that is not based on much testing 

and since this command is only used when needed, I don't think it will slow down the device. my assumption is that it will run only once while the device is booting up(with every Octoprint restart) and in some rare cases, if the domain name changes

Reply:
just pushed the change